### ISSUE TITLE
placeholder variable retaining state in MS VSCode

### ISSUE THREAD

**Issue by ValdarT** [machow/siuba#296]
`siuba` does not work in Microsoft Visual Studio Code interactive window: for some reason the placeholder variable retains its state (i.e., the outcome of the previous calculation) which of course prevents using it further (without reimporting).

I have version 1.1.3 of `pandas` and version 0.0.24 of `siuba`. 
From VScode side, I'm using 1.51.1 with v2020.11.0 of MS Python extension and v1.1.85 of Pyright extension installed.

Not sure whether there's anything you can do on the `siuba` side but wanted to let you know.

This is a fantastic initiative, thank you for this!

**Comment by machow** [comment]
Ah, thanks for the heads up! I haven't used VS Code in a while, but am downloading now to check it out :o.

**Comment by machow** [comment]
hmm.. I wasn't able to replicate, but am not too familiar with vs code. Are you using a specific extension to run code interactively in IPython?

Here's what I tried (but note that the interactive mode is using plain python, so I may be missing a critical piece..):

![vscode](https://user-images.githubusercontent.com/2574498/99466326-3923ee80-290a-11eb-87d2-115d054df515.gif)

**Comment by ValdarT** [comment]
Yes, it happens when run in [Interactive Window](https://code.visualstudio.com/docs/python/jupyter-support-py#_python-interactive-window). Running in terminal works fine for me as well.

**Comment by machow** [comment]
Ah, okay--thanks for the help. I tried running in a python interactive window, and it seems to work okay:

![python_interactive](https://user-images.githubusercontent.com/2574498/99687616-db4eee00-2a52-11eb-9ba2-4caee2531072.gif)

Any idea what I might be missing? If possible, would you mind sending a screencast (or screenshots) of the issue?

**Comment by ValdarT** [comment]
Hmm, no idea. The Pylance extension, perhaps?

![siuba_vscode_issue](https://user-images.githubusercontent.com/7242935/99721053-47fbd400-2ab7-11eb-936e-94810a05efac.gif)

**Comment by grst** [comment]
The underscore access to the last values is a feature of the IPython shell: 
https://ipython.org/ipython-doc/3/interactive/reference.html#output-caching-system

You could try setting `InteractiveShell.cache_size = 0` in your [IPython config file](https://ipython.org/ipython-doc/3/config/intro.html).
